library/ssl_tls.c

```diff
@@ -7357,6 +7357,7 @@ static int ssl_remember_peer_pubkey( mbedtls_ssl_context *ssl,
         return( MBEDTLS_ERR_SSL_INTERNAL_ERROR );
     }
 
+    ssl->handshake->got_peer_pubkey = 1;
```

Comment:
This line is missing a preprocessor guard of `#if !defined(MBEDTLS_SSL_KEEP_PEER_CERTIFICATE)`.

I'm not sure how we'd upstream, but if we did upstream this would need fixing.

Reply:
Good catch! Just curious, how did you find out (after the PR was merged)?